core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -284,12 +284,12 @@ trait Foldable[F[_]]  { self =>
     foldLeft1Opt(fa)(A.append(_, _))
 
   // should be revisited by https://github.com/scalaz/scalaz/issues/1663
-  /** map into an G[B] and pick the first "winner", as determined by Alternative */
-  def asumMap[A, B, G[_]](fa: F[A])(f: A => G[B])(implicit G: Alternative[G]): G[B] =
+  /** map into an G[B] and pick the first "winner", as determined by PlusEmpty */
+  def asumMap[A, B, G[_]](fa: F[A])(f: A => G[B])(implicit G: PlusEmpty[G]): G[B] =
     foldRight(fa, G.empty[B])((a, as) => G.plus(f(a), as))
 
-  /** pick the first "winner", as determined by Alternative */
-  def asum[G[_], A](fa: F[G[A]])(implicit G: Alternative[G]): G[A] =
+  /** pick the first "winner", as determined by PlusEmpty */
+  def asum[G[_], A](fa: F[G[A]])(implicit G: PlusEmpty[G]): G[A] =
     foldRight(fa, G.empty[A])(G.plus[A](_, _))
 
   def msuml[G[_], A](fa: F[G[A]])(implicit G: PlusEmpty[G]): G[A] =
```

Comment:
I prefer to use the Haskell name if one exists, as it makes it easier to find coming from Haskell and easier to use Haskell coming from scalaz. How about renaming `msuml` to `asuml` ?

Reply:
Haskell also has [`msum`](https://hackage.haskell.org/package/base-4.11.0.0/docs/Data-Foldable.html#g:4) which is the same as `asum`. I find both to be bad names: presumably, `m` stands for `MonadPlus` and `a` stands for `Alternative`, but neither `MonadPlus` nor `Alternative` is needed.

Reply:
That Haskell does something doesn't make it automatically good. Overly terse names are IMHO something it does badly.

Reply:
cats decided to rename everything to "beginner friendly" names and it's a disaster. In fact they seem to spend 99% of their time talking about the names of things. I prefer if we stick with Haskell names where there is a precedent, whether they are good names or bad names. A `final` forwarding method (or syntax) is a good compromise if there is a preferred secondary name for something that is better than the Haskell name.

Reply:
In this specific case, I would suggest `psum`, since it uses a _polymorphic_ monoid (`PlusEmpty`), as opposed to `sum`, which uses an ordinary monoid. It is a nice  coincidence that `PlusEmpty` also starts with a "p". And provide an alias named `asum`. WDYT?

Reply:
Renamed.